pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -58,9 +58,53 @@ func GetEnvironment(cr *v1.KieApp, service v1.PlatformService) (v1.Environment,
 	if err != nil {
 		return v1.Environment{}, err
 	}
+	mergedEnv, err = mergeDB(service, cr, mergedEnv, envTemplate)
+	if err != nil {
+		return v1.Environment{}, err
+	}
 	return mergedEnv, nil
 }
 
+func mergeDB(service v1.PlatformService, cr *v1.KieApp, env v1.Environment, envTemplate v1.EnvTemplate) (v1.Environment, error) {
+	dbEnvs := make(map[v1.DatabaseType]v1.Environment)
+	for i := range env.Servers {
+		kieServerSet := envTemplate.Servers[i]
+		if kieServerSet.Database.Type == "" {
+			continue
+		}
+		dbType := kieServerSet.Database.Type
+		if _, loadedDB := dbEnvs[dbType]; !loadedDB {
+			yamlBytes, err := loadYaml(service, fmt.Sprintf("envs/db-%s.yaml", dbType), cr.Namespace, envTemplate)
```

Comment:
might be better to segment the db configs in a dedicated dir... maybe `env/dbs/%s.yaml` ?
the configmap logic is already written as such that any additional subdirs will automatically create a new ConfigMap with an appropriate name.